### ISSUE TITLE
Docker Port Ranges

### ISSUE THREAD

**Issue by twllight** [docker/docker-py#1747]
Hello,

After looking at: https://github.com/docker/docker-py/pull/1072 Is it safe to assume that currently defining a range of ports to expose for a container is not implemented? if it is are there any examples that I could take a look at? Thank you.

**Comment by gtseres** [comment]
I don't think that this is supported, but you could do something like the following to construct the dict of ports. Obviously, you can skip the host_port_range parts if you want the container ports to be mapped to random host ports. Hope that helps.

```python
container_port_range = '2222-2232'
host_port_range = '3322-3332'

cstart, cend = map(int, container_port_range.split('-'))
crange = range(cstart, cend)
hstart, hend = map(int, host_port_range.split('-'))
hrange = range(hstart, hend)

assert crange == hrange

ports = {'{}/tcp'.format(c): h for c, h in zip(crange, hrange)}

>>> print ports
{'2231/tcp': 3331, '2225/tcp': 3325, '2222/tcp': 3322, '2223/tcp': 3323, '2229/tcp': 3329, '2226/tcp': 3326, '2227/tcp': 3327, '2224/tcp': 3324, '2230/tcp': 3330, '2228/tcp': 3328}
```

**Comment by gtseres** [comment]
@shin- Please let me know if you would find such a feature useful, and I 'll try to help.

**Comment by twllight** [comment]
thanks for this @gtseres I did something pretty similar to what you have which is using the range function.  I think the work being done in #1072 was good its just awaiting merging?

**Comment by gtseres** [comment]
@twllight Yeah, I guess so. Thanks for the finding.

**Comment by ijc** [comment]
Closing since it seems to have been addressed, according to my own investigation on #1072.
